src/scripts/tableActions.js

```diff
@@ -0,0 +1,40 @@
+import { createElement } from './utils';
+
+export function sortTableBody() {
+  let lastSortIndex = null;
+
+  return (tBody, column) => {
+    const sortedRows = [...tBody.rows].sort((rowA, rowB) => {
+      if (column < 3) {
+        const stringA = rowA.cells[column].textContent;
+        const stringB = rowB.cells[column].textContent;
+
+        return lastSortIndex !== column
+          ? stringA.localeCompare(stringB)
+          : stringB.localeCompare(stringA);
+      }
+
+      const numA = rowA.cells[column].textContent.replace(/[^0-9.-]+/g, '');
+      const numB = rowB.cells[column].textContent.replace(/[^0-9.-]+/g, '');
```

Comment:
The numeric values are extracted from the text content using a regular expression, but they are not converted to numbers before comparison. Ensure that `numA` and `numB` are parsed as numbers using `parseFloat` or `Number` to avoid string-based comparison.